spec/ics-007-tendermint-client/README.md

```diff
@@ -114,14 +126,20 @@ Tendermint client validity checking uses the bisection algorithm described in th
 function checkValidityAndUpdateState(
   clientState: ClientState,
   header: Header) {
-    // assert that header is newer than any we know
-    assert(header.height < clientState.latestHeight)
+    // assert trusting period has not yet passed
+    assert(currentTimestamp() - clientState.latestTimestamp < trustingPeriod)
+    // assert header timestamp is not in the future (& transitively that is not past the trusting period)
+    assert(header.timestamp <= currentTimestamp())
+    // assert header timestamp is past current timestamp
+    assert(header.timestamp > clientState.latestTimestamp)
```

Comment:
There still needs to be this check, correct?

```go
assert(header.timestamp - clientState.latestTimeStamp < trustingPeriod)
```

Reply:
That check is provided by lines 130 & 132. We need to check that the *local* time has not passed the trusting period because the header time could be falsified, checking the header time alone is not sufficient.

Reply:
Debugging issues in the relayer with @jackzampolin and @AdityaSripal.

132 introduces a fine clocked clock sync between the chains on each side of the connection.

You can replace `  assert(header.timestamp <= currentTimestamp())` with what Aditya proposed above `assert(header.timestamp - clientState.latestTimeStamp < trustingPeriod)`.

This will achieve the same clock sync with the same guarantees.